integration/integration_nexus_upload_test.go

```diff
@@ -17,115 +18,97 @@ import (
-	assert.NoError(t, err)
-	defer nexusContainer.Terminate(ctx)
-	ip, err := nexusContainer.Host(ctx)
-	assert.NoError(t, err)
-	port, err := nexusContainer.MappedPort(ctx, "8081")
-	assert.NoError(t, err, "Could not map port for nexus container")
-	nexusIpAndPort := fmt.Sprintf("%s:%s", ip, port.Port())
-	url := "http://" + nexusIpAndPort
-	resp, err := http.Get(url)
-	assert.Equal(t, resp.StatusCode, http.StatusOK)
+	container.assertHasFile(t, "/project/"+path.Base(url))
+}
 
-	cmd := command.Command{}
-	cmd.SetDir("testdata/TestNexusIntegration/mta")
+func TestNexus3UploadMta(t *testing.T) {
+	t.Parallel()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:       "sonatype/nexus3:3.25.1",
+		User:        "nexus",
+		TestDir:     []string{"testdata", "TestNexusIntegration", "mta"},
+		Environment: map[string]string{"NEXUS_SECURITY_RANDOMPASSWORD": "false"},
+		Setup: []string{
+			"/opt/sonatype/start-nexus-repository-manager.sh &",
+			"curl https://ftp.fau.de/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz | tar xz -C /tmp",
+			"echo PATH=/tmp/apache-maven-3.6.3/bin:$PATH >> ~/.profile",
+			"until curl --fail --silent http://localhost:8081/service/rest/v1/status; do sleep 5; done",
+		},
+	})
 
-	piperOptions := []string{
-		"nexusUpload",
-		"--groupId=mygroup",
-		"--artifactId=mymta",
-		"--user=admin",
-		"--password=admin123",
-		"--mavenRepository=maven-releases",
-		"--url=" + nexusIpAndPort,
+	err := container.whenRunningPiperCommand("nexusUpload", "--groupId=mygroup", "--artifactId=mymta",
+		"--user=admin", "--password=admin123", "--mavenRepository=maven-releases", "--url=http://localhost:8081")
+	if err != nil {
+		t.Fatalf("Piper command failed %s", err)
 	}
 
-	err = cmd.RunExecutable(getPiperExecutable(), piperOptions...)
-	assert.NoError(t, err, "Calling piper with arguments %v failed.", piperOptions)
+	container.assertHasOutput(t, "BUILD SUCCESS")
+	assertFileCanBeDownloaded(t, container, "http://localhost:8081/repository/maven-releases/mygroup/mymta/0.3.0/mymta-0.3.0.mtar")
+	assertFileCanBeDownloaded(t, container, "http://localhost:8081/repository/maven-releases/mygroup/mymta/0.3.0/mymta-0.3.0.yaml")
+}
 
-	cmd = command.Command{}
-	cmd.SetDir("testdata/TestNexusIntegration/maven")
+func TestNexus3UploadMaven(t *testing.T) {
+	t.Parallel()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:       "sonatype/nexus3:3.25.1",
+		User:        "nexus",
+		TestDir:     []string{"testdata", "TestNexusIntegration", "maven"},
+		Environment: map[string]string{"NEXUS_SECURITY_RANDOMPASSWORD": "false"},
+		Setup: []string{
+			"/opt/sonatype/start-nexus-repository-manager.sh &",
+			"curl https://ftp.fau.de/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz | tar xz -C /tmp",
+			"echo PATH=/tmp/apache-maven-3.6.3/bin:$PATH >> ~/.profile",
+			"until curl --fail --silent http://localhost:8081/service/rest/v1/status; do sleep 5; done",
```

Comment:
I tend to be sceptical about deduplication in tests, as the test should be in control of its own setup code. But I'm also not dogmatic. I'd keep it as is for now and see if it causes any issues.

Another idea I've had, but not yet implemented: We could also have a function that on the fly builds an image from a Dockerfile, then both tests could use a single dockerfile, and the setup would be at buildtime rather than at runtime. But honestly that feels like overengineering at this point. I think we improve the tests in readability, and also in debugability with this PR, and that is what I was going for.